internal/grader/grade.go

```diff
@@ -118,6 +133,25 @@ func Grade(ctx context.Context, assignment *model.Assignment, submissionPath str
 	return &gradingResult, nil, "", nil
 }
 
+// Resolve a proxy time for a given assignment.
+// If the proxy time is not specified, fall back to a day before the assignment due date.
+// If the assignment does not have a due date, the current time will be used for the proxy time.
+func ResolveProxyTime(proxyTime *timestamp.Timestamp, assignment *model.Assignment) *timestamp.Timestamp {
+	if proxyTime == nil {
+		if assignment.DueDate == nil {
+			now := timestamp.Now()
+			proxyTime = &now
+		} else {
+			// Convert time (one day) to hours -> minutes -> seconds -> msecs.
+			oneDay := int64(1 * 24 * 60 * 60 * 1000)
```

Comment:
I think using the time package here would be more clear:
`time.Duration(24 * time.Hour).Milliseconds()`.